File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        int n = nums.size();

         long long cnt = 0; 
         long long streak = 0;


         for ( auto num : nums){
            if(num == 0 ){
                streak ++;
                cnt = cnt + streak;
            }else{
                streak = 0;
            }
         }

         return (long long ) cnt;
    }
};

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: list[int]) -> int:
        total_subarrays = 0
        streak = 0
        
        for n in nums:
            if n == 0:
                streak += 1
                total_subarrays += streak
            else:
                streak = 0
                
        return total_subarrays